### DISCUSSION TITLE
An enhacement that I would really appreciate: the possibility of inserting empty pages

### DISCUSSION THREAD

**Discussion by BFH-ktt1** [stiglers-eponym/BeamerPresenter#33] [Ideas]
Dear developers,

I really enjoy your tool, great work. It would just be extremely valuable if one could insert "empty pages" in the pdf file to for example show some extra information or to explain/ask exercises. I do not know what the effort would be to implement this feature, but I would really appreciate it.

**Comment by stiglers-eponym** [comment]
Hi, this is a feature about which I have though a few times. For teaching I used the option to scroll down the page, which seemed simpler to implement. (The actions "scroll up", "scroll down" and "scroll normal" can be connected to buttons and keyboard shortcuts.)
At some point I want to implement a flexible mapping of slides to PDF pages (similar to Xournal++). Maybe I will find some time for this in a few weeks.

**Comment by stiglers-eponym** [comment]
Update: I implemented an incomplete test version of this feature in the `dev` branch. Features already implemented (but hardly tested) include inserting/removing slides and saving/loading drawings (limited compatibility with Xournal++). Some widgets (e.g. the thumbnail widget) still misbehave when inserting/removing slides.

In case you want to test this, use the actions "insert slide", "remove slide", and "restore slide". For example, in `gui.json` you can adjust the `tool selector` widget:
`{"type": "tool selector", buttons=[["insert slide", "remove slide", "restore slide"]]}`

**Reply by stiglers-eponym** [reply]
Update: this feature is included in version 0.2.6
